### ISSUE TITLE
Environment variables are not properly set

### ISSUE THREAD

**Issue by fgimenez** [CircleCI-Public/circleci-cli#391]
- [x] I have read [Contribution Guidelines](https://github.com/CircleCI-Public/circleci-cli/blob/master/CONTRIBUTING.md).
- [x] I have checked for similar issues and haven't found anything relevant.
- [x] This is not a security issue (which should be reported here: https://circleci.com/security/)

**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
When i set environment variables using `-e` on the command line they are not available on the build. When i execute:
```
circleci build --job test -e FOO1=BAR1 -e FOO2=BAR2
```
the env vars FOO1 and FOO2 are not accessible in the build, executing `env` as part of the build i can see:
```
[FOO1=BAR1,FOO2=BAR2]
DOCKER_VERSION=18.06.3
HOSTNAME=c48870140fea
SSH_AUTH_SOCK=/tmp/circleci-692488189/ssh_auth_sock
CIRCLE_WORKING_DIRECTORY=~/project
....
```
**What is the expected behavior?**
The variables set on the command line are available on the build.
**Which version of the CLI and OS are you using? Did this work in previous versions?**
```
$ circleci version
0.1.6949+d951526
$ circleci diagnostic
circleci diagnostic

---
CircleCI CLI Diagnostics
---
Debugger mode: false
Config found: /home/fgimenez/.circleci/cli.yml
API host: https://circleci.com
API endpoint: graphql-unstable
OK, got a token.
Trying an introspection query on API... 
Ok.
Hello, Federico Gimenez.
```

**Comment by nycnewman** [comment]
Think I see the same issue. Trying to pass in Docker and other credentials that are in CI context and work, through the circleci cli do not appear to show and get a docker login failure. Same version of cli as reporter.

Config works if I hardcode the variables in the yml. Fails when referenced in environment even if passed in on command line (or set in environment prior to cli execution)

**Comment by alexkli** [comment]
Same regression here with the latest version: `0.1.6949+d951526`.

If I downgrade to the previous version `0.1.6772+67c7d52` it works. Here is how I downgraded using Homebrew:

```
brew unlink circleci
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/2452da29fcc8da7f1dcf09a47eb149e1fcf57db3/Formula/circleci.rb
```

**Comment by tquach** [comment]
Seems like a duplicate of #387 and #385 and still no update from the CircleCI team.

**Comment by eserzomcity** [comment]
Proposed bug fix in https://github.com/CircleCI-Public/circleci-cli/pull/395

**Comment by brianthelion** [comment]
@marcomorain Status? I'm blocked on this:

```
====>> Spin Up Environment
Build-agent version  ()
Docker Engine Version: 19.03.6
Kernel Version: Linux f72755058628 5.3.0-46-generic #38-Ubuntu SMP Fri Mar 27 17:37:05 UTC 2020 x86_64 Linux
Starting container circleci/node:10.0-browsers
  image is cached as circleci/node:10.0-browsers, but refreshing...
10.0-browsers: Pulling from circleci/node
Digest: sha256:144fba119cd57b0ef1fa48a738c58408d38380650544019882f23ad555f1bd78
Status: Image is up to date for circleci/node:10.0-browsers
  using image circleci/node@sha256:144fba119cd57b0ef1fa48a738c58408d38380650544019882f23ad555f1bd78
====>> Preparing Environment Variables
Using build environment variables:
  BASH_ENV=/tmp/.bash_env-localbuild-1587661917
  CI=true
  CIRCLECI=true
  CIRCLE_BRANCH=master
  CIRCLE_BUILD_NUM=
  CIRCLE_JOB=build
  CIRCLE_NODE_INDEX=0
  CIRCLE_NODE_TOTAL=1
  CIRCLE_REPOSITORY_URL=https://github.com/f0cal/px4-firmware.git
  CIRCLE_SHA1=172e435ec01b8484b38e2c70bf30e25530f203ca
  CIRCLE_SHELL_ENV=/tmp/.bash_env-localbuild-1587661917
  CIRCLE_WORKING_DIRECTORY=~/project

Using environment variables from project settings and/or contexts:
  [MY_ENV_VAR=**REDACTED**
...
```
Note the `[` character. All supplied `--env` variables are affected.

**Comment by brianburnszd** [comment]
when trying @alexkli's proposed solution:

> Same regression here with the latest version: `0.1.6949+d951526`.
> 
> If I downgrade to the previous version `0.1.6772+67c7d52` it works. Here is how I downgraded using Homebrew:
> 
> ```
> brew unlink circleci
> brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/2452da29fcc8da7f1dcf09a47eb149e1fcf57db3/Formula/circleci.rb
> ```

In MacOS Catalina it was failing:

```
$ brew install https://raw.githubusercontent.com/...
Error: Calling Installation of circleci from a GitHub commit URL is disabled! Use 'brew extract circleci' to stable tap on GitHub instead.
``` 

I was able to install by doing this:

```
brew unlink circleci
cd $( brew --prefix )/Homebrew/Library/Taps/homebrew/homebrew-core
git checkout 2452da29fcc8da7f1dcf09a47eb149e1fcf57db3 Formula/circleci.rb
brew install circleci
```

Now `circleci local execute` works correctly.

**Comment by alexkli** [comment]
@brianburnszd Note that this issue has been fixed quite some time ago with #397 and the newest `circleci` cli should work again.

**Comment by jamesrweb** [comment]
> @brianburnszd Note that this issue has been fixed quite some time ago with #397 and the newest `circleci` cli should work again.

Still getting this error on the latest MacOS Big Sur actually. Installed via brew, ran `circleci local execute -c process.yml --job myjob` after setting up and the error throws. 

<img width="900" alt="Bildschirmfoto 2021-05-11 um 11 34 55" src="https://user-images.githubusercontent.com/47126579/117793845-ef7ae780-b24c-11eb-80e6-e3b6da65a8ff.png">

**Comment by Pomax** [comment]
> `brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/2452da29fcc8da7f1dcf09a47eb149e1fcf57db3/Formula/circleci.rb`

This is no longer possible, with brew erroring out, since this is insanely insecure. `Error: Installation of circleci from a GitHub commit URL is unsupported! `brew extract circleci` to a stable tap on GitHub instead.`
